### DISCUSSION TITLE
Collapsible History

### DISCUSSION THREAD

**Discussion by gabemaroz** [astashov/liftosaur#216] [Ideas]
# Collapsible History

## Problem
When the users history is quite long, often times there is too much information. Most people doing a quick scan want to find the same date as a previous program and then expand it.

## Solution

1. Add a new Setting with the header **Interace** which would include **Always On Display**, **Dark Mode** (for future updates), and **Collapse History** as a default. This could also be the place for any other user preferences of this type
2. Add a chevron which will collapse (⏵) or expand (⏷) the history entry. This is could be done by setting the `style="display: none;"` inside of `data-cy="history-entry"` for all but the most recent (aka upcoming) workout.
3. Clicking on the chevron would expand the history.

## Issues

* Currently click on the entry expands it into a full workout. This would need to be changed.
* Some users may not want their history collapsed. This is mitigated by the **Interface** setting as above. Probably need user feedback on which should be the default behavior.

## Caveats

* If the default is collapsed, it makes the current workout *very* easy to find.

## Mockup

<img width="571" alt="Screenshot 2024-11-11 at 12 21 51 PM" src="https://github.com/user-attachments/assets/49de4c08-97b9-4835-aca9-a27576f7aa1f">

**Comment by gabemaroz** [comment]
Three potential options:

- All: Collapse all history
- None: Same as current
- Except Previous: Keep the previous day uncollapsed (example, 4 day cycle, currently day 3, show previous day 3 of same cycle)

**Comment by astashov** [comment]
Hopefully it's less of an issue now with the calendar
